src/com/amazon/kinesis/streaming/agent/tailing/SourceFile.java

```diff
@@ -114,6 +115,13 @@ public String toString() {
     private void validateDirectory(Path dir) {
         Preconditions.checkArgument(dir != null, "Directory component is empty!");
         // TODO: validate that the directory component has no glob characters
+
+        //check the permissions of every level in the directory path
+        for(Path level : dir.toAbsolutePath()){
+            if(!Files.isReadable(level)){
```

Comment:
what are you trying to check here? do you want to check if the files are permitted to read by agent (the file grants r permission) or if the directory can be listed by the agent (the directory grants x permission)?